src/amo/containers/CategoryList.js

```diff
@@ -0,0 +1,30 @@
+import { compose } from 'redux';
+import { asyncConnect } from 'redux-connect';
+import { connect } from 'react-redux';
+
+import Categories from 'amo/components/Categories';
+import { loadCategoriesIfNeeded } from 'core/utils';
+
+
+export function mapStateToProps(state, ownProps) {
+  const addonType = ownProps.params.addonType.replace(/s$/, '');
```

Comment:
Is `addonType` translatable? Then the "always ends in an s" trick wouldn't work in other languages than English so the `externalAddonTypeMap` would be needed anyway.